src/core/server/plugins/legacy_config.ts

```diff
@@ -60,3 +70,33 @@ export const getGlobalConfig$ = (configService: IConfigService): Observable<Shar
     )
   );
 };
+
+// TODO: move this to Elasticsearch service itself
+const BEGIN_CERT_PRAGMA = `-----BEGIN CERTIFICATE-----\n`;
+const END_CERT_PRAGMA = `\n-----END CERTIFICATE-----`;
```

Comment:
This currently only supports PEM certs which are required to have these pragmas according to the spec. If there's a better way to handle this please let me know, because this doesn't feel great.

Reply:
I tested exporting a certificate from my Windows machine and it had CRLF endings (`\r\n`).
So I think it would be better to leave the line endings off of these pragmas and modify the parsing function accordingly.

As far as defining the pragmas: I think this is correct / the only way to do it.
Here is what node-forge does for PEM encoding and decoding, for reference: https://github.com/digitalbazaar/forge/blob/c0bb359afca73bb0f3ba6feb3f93bbcb9166af2e/lib/pem.js

Interestingly, for PEM decode it handles both LF and CRLF endings, but for encode it always uses CRLF endings.
I'm guessing that Node's [`X509Certificate` constructor](https://nodejs.org/docs/latest-v16.x/api/crypto.html#new-x509certificatebuffer) will gracefully handle a mix of LF and CRLF endings in a single PEM string, and it will behave the same on *nix and Windows, but it might be worth checking that too.